feat: Use prometheus setup task in integration test

bradbeam opened the pull request:
This changes out the postgres experiment with a dumb experiment.
This uses the approximateRuntime to do nothing, so we dont need to
be concerned with ci runner node resources.
This also configures a prometheus setup task so we can validate/verify that
it works as expected.

Signed-off-by: Brad Beam <brad.beam@stormforge.io>

bradbeam committed:
feat: Use prometheus setup task in integration test

This changes out the postgres experiment with a dumb experiment.
This uses the approximateRuntime to do nothing, so we dont need to
be concerned with ci runner node resources.
This also configures a prometheus setup task so we can validate/verify that
it works as expected.

Signed-off-by: Brad Beam <brad.beam@stormforge.io>

bradbeam commented on hack/experiment.yaml:
```diff
@@ -0,0 +1,108 @@
+      - args:
+        - prometheus
+        - $(MODE)
+        name: monitoring
```
Im debating adding in a trial pod here. 
Something like containers: [ locust, nginx ] so we could add a p95 latency metric just to increase the touch points. 

Thoughts?

jgustie replied:
I'll take you one further: should we use experiment generation here?

bradbeam committed:
feat: Use app generation in integration test

This improves integration test coverage:

- Covers app generation via `redskyctl generate experiment`
- Covers locust trial pod
- Covers prometheus metrics
- Covers push gateway usage

Signed-off-by: Brad Beam <brad.beam@stormforge.io>

jgustie commented on hack/integration.sh:
```diff
@@ -2,25 +2,36 @@
+
+echo "Create ci experiment"
+${REDSKYCTL_BIN} generate experiment -f hack/app.yaml > hack/experiment.yaml
+kubectl apply -f hack/experiment.yaml
```
How about this:

```
${REDSKYCTL_BIN} generate experiment -f hack/app.yaml --include-resources | tee experiment.yaml | kubectl apply -f -
${REDSKYCTL_BIN} generate trial -f experiment.yaml --default base | kubectl create -f -
```

I don't like leaving the Git repo in a dirty state though, I wonder if `${TMPDIR}/experiment.yaml` is safe or not. Alternately, you could drop the `tee` and `generate trial` could continue using `kubectl`, just via selector: `<(kubectl get experiment -s redskyops.dev/app=ci) -o yaml` since the actual experiment name is a pain.

jgustie commented on hack/integration.sh:
```diff
@@ -2,25 +2,36 @@
 
 set -e
 
+REDSKYCTL_BIN="${REDSKYCTL_BIN:=dist/redskyctl_linux_amd64/redskyctl}"
```
Would this be more useful for running locally if it were: `REDSKYCTL="${REDSKYCTL:-redskyctl}"` And then the CI action explicitly set the path to the full value?

bradbeam replied:
I went this way since the main use is for CI, but allows us to do `REDSKYCTL_BIN=$(which redskyctl) hack/integration.sh` locally ( what I did to test this :D )

jgustie commented on hack/locustfile.py:
```diff
@@ -0,0 +1,9 @@
+import time
+from locust import HttpUser, task, between
+
+class HomieUser(HttpUser):
```
Sup

jgustie commented on hack/nginx.yaml:
```diff
@@ -0,0 +1,39 @@
+            cpu: 50m
+---
+apiVersion: v1
+kind: Service
```
[Services first!](https://kubernetes.io/docs/concepts/cluster-administration/manage-deployment/#organizing-resource-configurations)

bradbeam committed:
chore: Minor cleanups from code review

Signed-off-by: Brad Beam <brad.beam@stormforge.io>

bradbeam committed:
chore: Add trap to for integration test

Hopefully help shed light when stuff breaks.

Signed-off-by: Brad Beam <brad.beam@stormforge.io>

bradbeam committed:
chore: Add ghz sleep, controller logs

Signed-off-by: Brad Beam <brad.beam@stormforge.io>

bradbeam commented on internal/experiment/generation/generation.go:
```diff
@@ -97,10 +97,16 @@ func loadApplicationData(app *redskyappsv1alpha1.Application, src string) ([]byt
 
 	opts := []getter.ClientOption{
 		func(c *getter.Client) error {
```
I can put this as a separate PR if needed since its unrelated to this. But without this change, we don't work.

jgustie replied:
This is fixed in the latest #288 changes, along with a comment clarifying why we _do not_ want use the process working directory here.

jgustie commented on hack/integration.sh:
```diff
@@ -2,25 +2,53 @@
-dist/redskyctl_linux_amd64/redskyctl init --wait
-echo "Create postgres experiment"
-kustomize build hack | kubectl apply -f -
+${REDSKYCTL_BIN} init
```
Why did you remove the `--wait` here?

bradbeam replied:
I can't seem to find the failed build, but I believe it was due to randomly getting an exit 1.
I'm assuming it's the same reason I had to add in the sleep below where the resources in Kubernetes weren't available/present by the time we got to waiting on the resource.

jgustie commented on hack/integration.sh:
```diff
@@ -2,25 +2,53 @@
+
+Stats() {
+  ${REDSKYCTL_BIN} generate experiment -f hack/app.yaml
+	kubectl get po -o wide
```
Tab, also nit, since this is a script why abbreviate `pods` to `po`?

bradbeam replied:
probably just habit, will update

bradbeam committed:
chore: Cleanup paths for app.yaml

Signed-off-by: Brad Beam <brad.beam@stormforge.io>
